src/services/TestResultsService.js

```diff
@@ -16,64 +17,64 @@ class TestResultsService {
+        }).catch(error => {
+          if (!(error instanceof HTTPError)) {
+            console.error(error)
+            error = new HTTPError(500, 'Internal Server Error')
           }
-        }
-        testResults = this.removeTestResultId(testResults)
-        testResults = testResults.map((testResult) => this.removeVehicleClassification(testResult))
-        return testResults
-      })
-      .catch(error => {
-        if (!(error instanceof HTTPError)) {
-          console.error(error)
-          error = new HTTPError(500, 'Internal Server Error')
-        }
-        throw error
-      })
-  }
-
-  removeTestResultId (testResults) {
-    if (testResults) {
-      for (let i = 0; i < testResults.length; i++) { delete testResults[i].testResultId }
+          throw error
+        })
+      } else if (filters.testerStaffId) {
+        return this.testResultsDAO.getByTesterStaffId(filters.testerStaffId).then(data => {
+          return this.checkDAOResponse(data, filters)
+        }).catch(error => {
+          if (!(error instanceof HTTPError)) {
+            console.error(error)
+            error = new HTTPError(500, 'Internal Server Error')
+          }
+          throw error
+        })
+      } else {
+        return Promise.reject(new HTTPError(400, 'Bad request'))
+      }
+    } else {
+      return Promise.reject(new HTTPError(400, 'Bad request'))
     }
-    return testResults
   }
-
-  filterTestResultsByStatus (testResults, status) {
-    var filteredTestResults = testResults.filter(
-      function (testResult) { return testResult.testStatus === status }
-    )
-    return filteredTestResults
+  checkTestResults (data) {
+    if (data) {
+      if (!data.Count) {
+        throw new HTTPError(404, 'No resources match the search criteria')
+      }
+    }
+    return data.Items
   }
 
-  filterTestResultByDate (testResults, fromDateTime, toDateTime) {
-    for (let i = 0; i < testResults.length; i++) {
-      testResults[i].testTypes = testResults[i].testTypes.filter(
-        function (testType) {
-          return (!(dateFns.isAfter(testType.createdAt, toDateTime) || dateFns.isBefore(testType.createdAt, fromDateTime)))
-        })
+  checkDAOResponse (data, filters) {
+    let testResults
+    testResults = this.checkTestResults(data)
+    testResults = GetTestResults.filterTestResultByDate(testResults, filters.fromDateTime, filters.toDateTime)
+    if (filters.testStatus) {
+      testResults = GetTestResults.filterTestResultsByParam(testResults, 'testStatus', filters.testStatus)
     }
-    testResults = testResults.filter(
-      function (testResult) {
-        return testResult.testTypes.length !== 0
-      })
+    if (filters.testStationPNumber) {
+      testResults = GetTestResults.filterTestResultsByParam(testResults, 'testStationPNumber', filters.testStationPNumber)
+    }
+    if (testResults.length === 0) {
+      return new HTTPError(404, 'No resources match the search criteria')
```

Comment:
Shouldn't this error be caught in `getTestResults` and also thrown?

Reply:
Done